### PR TITLE
FAD-6181 added sending domains tab, fixing bug in api keys tab, need …

### PR DESCRIPTION
**Pull request by bizob2828**
I had to abstract some sending domains components so they can be shared on list and then the sending domains tab in sending domains edit

I ended up fixing two bugs while testing things:
 * Editing an API key from subaccounts details tab
 * Creating a subaccount with ip pool(was passing nothing, edit was ok)
 * I tried to add tooltip to grants but CSS is f'ed. I asked jon to help and if it's simple we'll keep otherwise I'll restore
 * I also removed duplicate grantCheckboxes code and updated so both places use one component
 * there was some PermissionsCheckList that was also not being used anywhere, but it had tests!

### CODE REVIEW

File: src/pages/subaccounts/DetailsPage.js [hunk 1]
```diff
@@ -41,8 +61,9 @@ export class DetailsPage extends Component {
   }
 
   render() {
-    const { loading, location, tabs } = this.props;
-    const selectedTab = location.pathname.endsWith('keys') ? 1 : 0;
+    const { loading, location, id } = this.props;
+    const tabs = buildTabs(id);
+    const selectedTab = tabs.findIndex(({ to }) => location.pathname === to);
```
Comment (bizob2828) [inline src/pages/subaccounts/DetailsPage.js]: write some tests for this

File: src/components/grantBoxes/GrantsCheckboxes.js [hunk 2]
```diff
@@ -12,15 +12,16 @@ const cx = classnames.bind(styles);
...
-    <Field
-      key={grant.key}
-      name={`grants[${grant.key}]`}
-      label={grant.label}
-      component={CheckboxWrapper}
-      type="checkbox"
-      normalize={(value) => (value ? 'true' : '')} // TODO is this necessary? otherwise we get proptype warnings
-    />
+    // TODO: fix css
+    <Tooltip key={grant.key} dark content={grant.description}>
```
Comment (bizob2828) [inline src/components/grantBoxes/GrantsCheckboxes.js]: tried to do this but css is f'ed. I can revert unless someone has a clever fix _hint_ @jonambas _hint_

Comment (jonambas) [reply src/components/grantBoxes/GrantsCheckboxes.js]: I'd check MetricsModal in the summary report and maybe try the same css

File: src/components/reduxFormWrappers.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ export const CheckboxWrapper = ({ input, meta, ...rest }) => {
     <Checkbox
       id={input.name}
       {...input}
-      checked={input.value}
+      checked={input.value || false}
```
Comment (bizob2828) [inline src/components/reduxFormWrappers.js]: i can fix this by tweaking the initial values of the grantsRadio

File: src/selectors/subaccounts.js [hunk 4]
```diff
@@ -14,6 +13,7 @@ const formatSubaccount = ({ compliance_status = 'active', status = 'active', ...
 };
 
 export const hasSubaccounts = (state) => state.currentUser.has_subaccounts;
+export const getSubaccount = (state) => state.subaccount;
```
Comment (bizob2828) [inline src/selectors/subaccounts.js]: we don't need that

File: src/pages/subaccounts/DetailsPage.js [hunk 5]
```diff
@@ -5,12 +5,13 @@ import { Page, Tabs } from '@sparkpost/matchbox';
...
 import { getSubaccount } from 'src/actions/subaccounts';
 import { ApiKeySuccessBanner } from 'src/components';
-import { selectSubaccount, selectDetailTabs } from 'src/selectors/subaccounts';
+import { selectSubaccount } from 'src/selectors/subaccounts';
 import { listPools } from 'src/actions/ipPools';
 import { listApiKeys, hideNewApiKey } from 'src/actions/api-keys';
 
 import ApiKeysTab from './components/ApiKeysTab';
 import EditTab from './components/EditTab';
+import SendingDomainsTab from './components/SendingDomainsTab';
```
Comment (gwalls) [inline src/pages/subaccounts/DetailsPage.js]: As discussed irl, load sending domains on `componentDidMount` of the `DetailsPage`

**Pull request closed without merging**